tests/functionnal/Ticket.php

```diff
@@ -1389,6 +1389,46 @@ public function testUpdateFollowup() {
          ->variable['users_id_editor']->isEqualTo($uid2);
    }
 
+   public function testClone() {
+      $this->login();
+      $this->setEntity('Root entity', true);
+      $ticket = new \Ticket();
+      $ticket = getItemByTypeName('Ticket', '_ticket01');
+
+      $date = date('Y-m-d H:i:s');
+      $_SESSION['glpi_currenttime'] = $date;
+
+      // Test item cloning
+      $added = $ticket->clone();
+      $this->integer((int)$added)->isGreaterThan(0);
+
+      $clonedTicket = new \Ticket();
+      $this->boolean($clonedTicket->getFromDB($added))->isTrue();
+
+      $fields = $ticket->fields;
+
+      // Check the ticket values. Id and dates must be different, everything else must be equal
+      foreach ($fields as $k => $v) {
+         switch ($k) {
+            case 'id':
+               $this->variable($clonedTicket->getField($k))->isNotEqualTo($ticket->getField($k));
+               break;
+            case 'date_mod':
+            case 'date_creation':
+               $dateClone = new \DateTime($clonedTicket->getField($k));
+               $expectedDate = new \DateTime($date);
+               $this->dateTime($dateClone)->isEqualTo($expectedDate);
+               break;
+            default:
+               $this->executeOnFailure(
```

Comment:
Oh, I did not know this one! :)

Reply:
Oh I don't know if I should keep it however, but it helped me debug some failing tests.

Reply:
Since this may provide extra info in case of test failure, you can keep it ;) Also, atoum propose a `dump()` method that may be more "clean" than a `var_dump`; but I do not remember exactly how it works (should be ok with the `--debug` flag which is set on CI).